### PR TITLE
Change tracking to track granted only when it's a new state

### PR DESCRIPTION
**Pull request by janua**
This fixes the tracking so that it will only count `granted` state if it has not been granted before.

@samdesborough @NathanielBennett

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/notifications.js [hunk 1]
```diff
@@ -107,9 +107,10 @@ define([
         },
 
         subscribeHandler: function () {
+            var alreadyGranted = Notification.permission === 'granted';
             modules.subscribe().then(modules.follow)
                 .then(function() {
-                    if (Notification.permission === 'granted') {
+                    if (!alreadyGranted && Notification.permission === 'granted') {
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: is `Notification.permission` mutated by subscribing/following?

Comment (janua) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: Yes... Is that going to be a problem?

**Comment by michaelwmcnamara** [comment]
Worth holding off till Monday?

**Comment by janua** [comment]
@michaelmcnamara No one will be deploying anything today.

File: static/src/javascripts/bootstraps/enhanced/notifications.js (hunk 1 above)
Comment (desbo) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: it does make this code a bit of a weird read,
I had a quick look if there's an event that's fired when `Notification.permission` changes, but there doesn't seem to be one. 
I'd probably just add a comment around here that explains it gets changed at some point during `modules.subscribe` or `modules.follow`

Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/notifications.js]: is it `if (wasNotGranted && isNowGranted) {}`, is that the logic?

**Comment by janua** [comment]
@sndrs That's exactly the logic. I renamed the `var`s to make it more clear; maybe I shouldn't have the superfluous `var` though?

**Comment by janua** [comment]
Also added a comment closer to where the permission gets asked.

**Comment by sndrs** [comment]
👍 :beautiful:

**Pull request merged**